rust/lance-index/src/scalar/inverted.rs

```diff
@@ -233,19 +237,29 @@ impl ScalarIndex for InvertedIndex {
 // it also records the frequency of each token
 #[derive(Debug, Clone, Default, DeepSizeOf)]
 struct TokenSet {
-    tokens: Vec<String>,
-    ids: Vec<u32>,
-    frequencies: Vec<u64>,
+    // token -> (token_id, frequency)
+    tokens: HashMap<String, (u32, u64)>,
+    next_id: u32,
 }
 
 impl TokenSet {
     fn to_batch(&self) -> Result<RecordBatch> {
-        let token_col = StringArray::from(self.tokens.clone());
-        let token_id_col = UInt32Array::from(self.ids.clone());
-        let frequency_col = UInt64Array::from(self.frequencies.clone());
+        let mut tokens_builder = LargeStringBuilder::with_capacity(self.tokens.len(), 32);
+        let mut token_id_builder = UInt32Builder::with_capacity(self.tokens.len());
+        let mut frequency_builder = UInt64Builder::with_capacity(self.tokens.len());
```

Comment:
Since these aren't nullable, I think the faster approach would be to use a `Vec`, and then convert from a vec into the appropriate array at the end. This could say some cycles involved in handling the null buffer.

Reply:
can arrow reuse the vector's data?
here with the array builder I think it doesn't need to copy the data from builder to array again.
but for vector it does need right?

Reply:
Arrays can re-use the vectors data without copying.